### PR TITLE
Add end session endpoint

### PR DESCRIPTION
**Pull request by jonalu**
> First version of the SLO flow: https://difi.github.io/felleslosninger/oidc_func_sso.html#initiering-av-slo-session-management

## Work
- Add `end_session_endpoint` to the `.well-known` data structure
- Add handler for `end_session_endpoint` that redirects to the url defined in `req.query.post_logout_redirect_uri`
- Updated the README to specify port `8080`, not `1111`

## Future work
- The `post_logout_redirect_uri` should be resolved from the `Issuer` instance
- Implement validation and other return params defined in the difi SLO flow ☝️

### CODE REVIEW

File: src/main/kotlin/no/nav/security/mock/oauth2/http/OAuth2HttpRequestHandler.kt [hunk 1]
```diff
@@ -88,6 +90,12 @@ class OAuth2HttpRequestHandler(
                     val tokenRequest: TokenRequest = request.asTokenRequest()
                     json(grantHandler(tokenRequest).tokenResponse(tokenRequest, request.url.toIssuerUrl(), oAuth2TokenCallback))
                 }
+                url.isEndSessionEndpointUrl() -> {
+                    // TODO: The redirect should be to Issuer.post_logout_redirect_uri
+                    log.debug("handle end session request $request")
+                    val postLogoutRedirectUri = request.url.queryParameter("post_logout_redirect_uri") ?: "http://nav.no"
```
Comment (tronghn) [inline src/main/kotlin/no/nav/security/mock/oauth2/http/OAuth2HttpRequestHandler.kt:96]:
```suggestion
                    val postLogoutRedirectUri = request.url.queryParameter("post_logout_redirect_uri") ?: "https://www.nav.no"
```

File: src/main/kotlin/no/nav/security/mock/oauth2/extensions/HttpUrlExtensions.kt [hunk 2]
```diff
@@ -7,13 +7,15 @@ import okhttp3.HttpUrl
 fun HttpUrl.isWellKnownUrl(): Boolean = this == this.toWellKnownUrl() || this == this.toOAuth2AuthorizationServerMetadataUrl()
 fun HttpUrl.isAuthorizationEndpointUrl(): Boolean = this.withoutQuery() == this.toAuthorizationEndpointUrl()
 fun HttpUrl.isTokenEndpointUrl(): Boolean = this == this.toTokenEndpointUrl()
+fun HttpUrl.isEndSessionEndpointUrl(): Boolean = this.toString().contains(this.toEndSessionEndpointUrl().toString(), ignoreCase = true)
```
Comment (tronghn) [inline src/main/kotlin/no/nav/security/mock/oauth2/extensions/HttpUrlExtensions.kt:10]:
```suggestion
fun HttpUrl.isEndSessionEndpointUrl(): Boolean = this.withoutQuery() == this.toEndSessionEndpointUrl()
```

**Review by tronghn — CHANGES_REQUESTED** [review]
Nice! Could perhaps add some more basic tests in`MockOAuth2ServerTests` (e.g. in `assertWellKnownResponseForIssuer()`)

File: src/main/kotlin/no/nav/security/mock/oauth2/extensions/HttpUrlExtensions.kt (hunk 2 above)
Comment (jonalu) [reply src/main/kotlin/no/nav/security/mock/oauth2/extensions/HttpUrlExtensions.kt:10]: ✅

**Comment by jonalu** [comment]
Thanks @tronghn
Do you mind having another look?

**Review by tronghn — APPROVED** [review]
lgtm

**Pull request merged**